mpas_analysis/test/test_timekeeping.py

```diff
@@ -0,0 +1,200 @@
+        date2 = MpasRelativeDelta(years=0, months=0, days=0, hours=0,
+                                  minutes=0, seconds=5, calendar=calendar)
+        self.assertEqual(date1, date2)
+
+        date1 = stringToDatetime('1996-01-15')
+        date2 = stringToRelativedelta('0005-00-00',
+                                      calendar='gregorian')
+        diff = date1-date2
+        self.assertEqual(diff, stringToDatetime('1991-01-15'))
+
+        date1 = stringToDatetime('1996-01-15')
+        date2 = stringToRelativedelta('0000-02-00',
+                                      calendar='gregorian')
+        diff = date1-date2
+        self.assertEqual(diff, stringToDatetime('1995-11-15'))
+
+        date1 = stringToDatetime('1996-01-15')
+        date2 = stringToRelativedelta('0000-00-20',
+                                      calendar='gregorian')
+        diff = date1-date2
+        self.assertEqual(diff, stringToDatetime('1995-12-26'))
+
+        # since pandas and xarray use the numpy type 'datetime[ns]`, which
+        # has a limited range of dates, the date 0001-01-01 gets increased to
+        # the minimum allowed year boundary, 1678-01-01 to avoid invalid
+        # dates.
+        date1 = clampToNumpyDatetime64(stringToDatetime('0001-01-01'),
+                                       yearOffset=0)
+        date2 = datetime.datetime(year=1678, month=1, day=1)
+        self.assertEqual(date1, date2)
+
+        date1 = clampToNumpyDatetime64(stringToDatetime('0001-01-01'),
+                                       yearOffset=1849)
+        date2 = datetime.datetime(year=1850, month=1, day=1)
+        self.assertEqual(date1, date2)
+
+        # since pandas and xarray use the numpy type 'datetime[ns]`, which
+        # has a limited range of dates, the date 9999-01-01 gets decreased to
+        # the maximum allowed year boundary, 2262-01-01 to avoid invalid
+        # dates.
+        date1 = clampToNumpyDatetime64(stringToDatetime('9999-01-01'),
+                                       yearOffset=0)
+        date2 = datetime.datetime(year=2262, month=1, day=1)
+        self.assertEqual(date1, date2)
+
+        # test if the calendars behave as they should
+        self.assertEqual(stringToDatetime('2016-02-28') +
+                         stringToRelativedelta('0000-00-01',
+                                               calendar='gregorian'),
+                         stringToDatetime('2016-02-29'))
+
+        self.assertEqual(stringToDatetime('2016-02-28') +
+                         stringToRelativedelta('0000-00-01',
+                                               calendar='gregorian_noleap'),
+                         stringToDatetime('2016-03-01'))
+
+        self.assertEqual(stringToDatetime('2016-03-01') -
+                         stringToRelativedelta('0000-00-01',
+                                               calendar='gregorian'),
+                         stringToDatetime('2016-02-29'))
```

Comment:
How about simplifying via something like:

```python
for calendar, expected in zip(['gregorian', 'gregorian_noleap'], ['2016-03-01', '2016-02-29']):
    self.assertEqual(stringToDateTime('2016-03-01') - stringToRelativedelta('0000-00-01', calendar=calendar),
                                stringToDateTime(expected))
```
This more clearly shows the distinction between the calendars and their usage and reduces the amount of needed code.